### PR TITLE
support create ssh connection by using sshkey (issue #1950)

### PR DESCRIPTION
**Pull request by tyusr**
see issue #1950

### CODE REVIEW

File: tools/nni_cmd/tensorboard_utils.py [hunk 1]
```diff
@@ -42,7 +42,8 @@ def copy_data_from_remote(args, nni_config, trial_content, path_list, host_list,
         local_path = os.path.join(temp_nni_path, trial_content[index].get('id'))
         local_path_list.append(local_path)
         print_normal('Copying log data from %s to %s' % (host + ':' + path_list[index], local_path))
-        sftp = create_ssh_sftp_client(host, machine_dict[host]['port'], machine_dict[host]['username'], machine_dict[host]['passwd'])
+        sftp = create_ssh_sftp_client(host, machine_dict[host]['port'], machine_dict[host]['username'], machine_dict[host]['passwd'],
+                                      machine_dict[host]['sshKeyPath'], machine_dict[host]['passphrase'])
```
Comment (SparkSnail) [inline tools/nni_cmd/tensorboard_utils.py]: what if `machine_dict[host]` do not have `sshKeyPath` field?

Comment (tyusr) [reply tools/nni_cmd/tensorboard_utils.py]: Oh sorry...

How about modify line 40
https://github.com/microsoft/nni/blob/6b375d6d1e402bad020ac213edf39aec6ad7b11d/tools/nni_cmd/tensorboard_utils.py#L39-L46

to 
```python
machine_dict[machine['ip']] = {'port': machine['port'], 'passwd': machine['passwd'], 'username': machine['username'], 'sshKeyPath': machine.get('sshKeyPath'), 'passphrase': machine.get('sshKeyPath')}
```

Comment (SparkSnail) [reply tools/nni_cmd/tensorboard_utils.py]: sure

Comment (tyusr) [reply tools/nni_cmd/tensorboard_utils.py]: > sure

done.

File: tools/nni_cmd/tensorboard_utils.py [hunk 2]
```diff
@@ -37,12 +37,14 @@ def copy_data_from_remote(args, nni_config, trial_content, path_list, host_list,
     machine_dict = {}
     local_path_list = []
     for machine in machine_list:
-        machine_dict[machine['ip']] = {'port': machine['port'], 'passwd': machine['passwd'], 'username': machine['username']}
+        machine_dict[machine['ip']] = {'port': machine['port'], 'passwd': machine['passwd'], 'username': machine['username'],
+                                       'sshKeyPath': machine.get('sshKeyPath'), 'passphrase': machine.get('sshKeyPath')}
```
Comment (SparkSnail) [inline tools/nni_cmd/tensorboard_utils.py]: `'passphrase': machine.get('sshKeyPath')`  --> `'passphrase': machine.get('passphrase')`

File: tools/nni_cmd/ssh_utils.py [hunk 3]
```diff
@@ -30,12 +30,16 @@ def copy_remote_directory_to_local(sftp, remote_path, local_path):
...
-def create_ssh_sftp_client(host_ip, port, username, password):
+def create_ssh_sftp_client(host_ip, port, username, password, ssh_key_path, passphrase):
     '''create ssh client'''
     try:
         paramiko = check_environment()
         conn = paramiko.Transport(host_ip, port)
-        conn.connect(username=username, password=password)
+        if ssh_key_path is not None:
+            ssh_key = paramiko.from_private_key_file(ssh_key_path, password=passphrase)
+            conn.connect(username=username, password=password, pkey=ssh_key)
```
Comment (SparkSnail) [inline tools/nni_cmd/ssh_utils.py]: If users set pkey, the password will be None?

**Pull request merged**